platform-sdk/docs/proposals/TSS-Library/tss/TssService.java

```diff
@@ -0,0 +1,160 @@
+import com.swirlds.signaturescheme.api.PairingPrivateKey;
+import com.swirlds.signaturescheme.api.PairingPublicKey;
+import com.swirlds.signaturescheme.api.PairingSignature;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.List;
+
+/**
+ * A Threshold Signature Scheme Service.
+ * <p>
+ * Contract of TSS:
+ * <ul>
+ *     <li>Generate TSSMessages out of PrivateShares</li>
+ *     <li>Verify TSSMessages out of a ParticipantDirectory</li>
+ *     <li>Obtain PrivateShares out of TssMessages for each owned share</li>
+ *     <li>Aggregate PrivateShares</li>
+ *     <li>Obtain PublicShares out of TssMessages for each share</li>
+ *     <li>Aggregate PublicShares</li>
+ *     <li>sign Messages</li>
+ *     <li>verify Signatures</li>
+ *     <li>Aggregate Signatures</li>
+ * </ul>
+ * @implNote an instance of the service would require a source of randomness {@link java.util.Random}, and a{@link com.swirlds.signaturescheme.api.SignatureSchema}
+ */
+public interface TssService {
+
+    /**
+     * Verify that the message is valid.
+     *
+     * @param pendingParticipantDirectory the participant directory used to generate the message
+     * @return true if the message is valid, false otherwise
+     */
+     boolean verifyTssMessage(@NonNull final ParticipantDirectory participantDirectory, @NonNull final TssMessage message);
+
+    /**
+     * Generate a TSS message for a pendingParticipantDirectory, for each private share owned.
+     *
+     * @param pendingParticipantDirectory the pending participant directory that we should generate the message for
+     * @param privateShare       the secret to use for generating new keys
+     * @return a TSSMessage for each owned share.
+     */
+    @NonNull
+    List<TssMessage> generateTssMessages(
+            @NonNull final ParticipantDirectory pendingParticipantDirectory,
+            @NonNull final TssPrivateShare privateShare);
+
+    /**
+     * Generate a TSS message for a pendingParticipantDirectory, from a random private share.
+     *
+     * @param pendingParticipantDirectory the pending participant directory that we should generate the message for
+     * @param privateShare       the secret to use for generating new keys
+     * @return a TSSMessage produced out of a random share.
+     */
+    @NonNull
+    TssMessage generateTssMessage(@NonNull final ParticipantDirectory pendingParticipantDirectory);
+
+    /**
+     * Compute all private shares that belongs to this participant.
+     *
+     * @param pendingParticipantDirectory the pending participant directory that we should generate the message for
+     * @param validTssMessages       the TSS messages to extract the private shares from. They must be previously validated.
+     * @return the list of private share the current participant owns, or null if there aren't enough shares to meet the threshold
+     */
+    @Nullable
+    List<TssPrivateShare> decryptPrivateShares(
```

Comment:
Does the order of these matter?

Reply:
The order of the list matches the order of the shareids belonging to the Participant. 
Order does not really matter, as the shareId is also a property in the object. It is just for consistency.
Will add this to the javadoc.